src/components/CreateTaskForm.tsx

```diff
@@ -0,0 +1,54 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { v4 as uuidv4 } from "uuid";
+import { taskSchema } from "../config/validation/taskSchema";
+import { useTasksStore } from "../stores/tasks";
+import { TaskFormData, TasksStore } from "../types/Task";
+import TaskFormLayout from "./TaskFormLayout";
+
+export default function CreateTaskForm({ onClose }: { onClose: () => void }) {
+  const addTask = useTasksStore((state: TasksStore) => state.addTask);
+  const id = uuidv4();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<TaskFormData>({
+    resolver: zodResolver(taskSchema),
+  });
+
+  const onSubmit = (data: TaskFormData) => {
+    const newTask = {
+      id: id,
+      title: data.title,
+      priority: data.priority,
+      storyPoints: data.storyPoints,
+      assignee: data.assignee,
+      dueDate: data.dueDate
+    };
+
+    const existingTasks = JSON.parse(localStorage.getItem("tasks") || "[]");
+    const updatedTasks = [...existingTasks, newTask];
+    localStorage.setItem("tasks", JSON.stringify(updatedTasks));
+
+    addTask(newTask);
+
+    reset();
```

Comment:
I think you can define a helper function called onSuccess and wrap this two functions there

Reply:
That's just what I thought, actually there is still some refactor I want to do